addons/stock_account/models/product.py

```diff
@@ -8,17 +8,55 @@
 from collections import defaultdict
 from datetime import datetime
 
+from odoo.addons.stock_account import const
+
 
 class ProductTemplate(models.Model):
     _inherit = ['product.template']
 
-    cost_method = fields.Selection(related="categ_id.property_cost_method", readonly=True)
-    valuation = fields.Selection(related="categ_id.property_valuation", readonly=True)
+    cost_method = fields.Selection(
+        string="Cost Method",
+        selection=const.COST_METHOD,
+        compute='_compute_cost_method',
+    )
+    valuation = fields.Selection(
+        string="Valuation",
+        selection=const.VALUATION,
+        compute='_compute_valuation',
+    )
     lot_valuated = fields.Boolean(
         "Valuation by Lot/Serial number",
         help="If checked, the valuation will be specific by Lot/Serial number.",
     )
 
+    @api.depends_context('company')
+    @api.depends("categ_id.property_cost_method")
+    def _compute_cost_method(self):
+        for product_template in self:
+            product_template.cost_method = (
+                product_template.categ_id.with_company(
+                    product_template.company_id
+                ).property_cost_method
+                or (product_template.company_id or self.env.company).cost_method
+            )
+
+    @api.depends_context('company')
+    @api.depends('categ_id.property_valuation')
+    def _compute_valuation(self):
+        pt_with_category = self.filtered('categ_id')
+        has_group_stock_accounting_automatic = self.env.user.has_group(
+            'stock_account.group_stock_accounting_automatic'
```

Comment:
this is not related to any company record. I would add a new field `valuation` on company too to be more flexible

Reply:
Hello @Whenrow 

I am not really sure about adding valuation in company to get default valuation because if group is enabled in settings means user want to use automatic valuation in their company and default will be automatic and if user want manual valuation on some product then user can change valuation on some categories to manual valuation and use that category in those products
As it was mention in specs to do that way to get default value depending on group value

Also I am not really getting good feeling about adding four field on company form view just to make category not require. Income and Expense account are okay as they are generic and used in different places but `Cost Method` and `Valuation` are specific to stock

May be we should move Cost Method in settings to get default value and check about valuation with DALA if we really want to add another option in setting or company to get default value as IMO `group_stock_accounting_automatic` Group is sufficient as valuation only contain two selection

Can you check @dalaOdoo and @Feyensv ?

Have a nice day!

Reply:
No strong opinion on the subject, I don't mind adding a field on `res.company` to allow a more precise management of the setting depending on the company (instead of the global group).  As it's the scope of the logistics team, if they believe it's better like this, we can add it.

Reply:
No need for a company setting, if no category on the product the valuation should be "manual"

Reply:
@dalaOdoo And if group is enabled `group_stock_accounting_automatic` then `automatic` ?

Reply:
no never automatic